rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java

```diff
@@ -113,6 +114,33 @@ public void putBlocks(long blockNumber, List<IndexedBlockStore.BlockInfo> blocks
         index.put(blockNumber, blocks);
     }
 
+    @Override
+    public void removeBlock(long blockNumber, Keccak256 blockHash) {
+        if (!index.containsKey(blockNumber)) {
```

Comment:
not sure we need this extra check.. it's basically two db queries - you already do same when checking `binfos` for `null` below

Reply:
that's true